sympy/solvers/tests/test_recurr.py

```diff
@@ -184,6 +184,10 @@ def test_rsolve():
             (C0*((sqrt(-a**2 + 1) - 1)/a)**n +
              C1*((-sqrt(-a**2 + 1) - 1)/a)**n)).simplify() == 0
 
+    assert rsolve((k + 1)*y(k), y(k)) is None
+    assert (rsolve((k + 1)*y(k) + (k + 3)*y(k + 1) + (k + 5)*y(k + 2), y(k))
+            is None)
```

Comment:
> Well, SymPy does this a lot. Whenever it's not able to compute say an integral. It returns the expr as it is.

That's basically an unevaluated object it returns, when it doesn't knows how to further process (evaluate) the input.

say for e.g.
```python
def foo(user_input = i):
    if i is this1:
        do this1
    if i is this2:
        do this2:
            while doing this2:
                if this_x fails: raise NotImplementedError
    if i is this3:
        do this
    else
        return Foo(i)  # unevaluated object
```

Otherwise it raises `ValueError` if input is invalid, as per my understanding.

> Well, we can raise an exception. LIke ValueError("Unable to compute degree")?

Maybe, but since I am unaware of the algorithm, so I am not sure if it's the right candidate.